dialogs/FaceEditSheet.cpp

```diff
@@ -0,0 +1,502 @@
+#include "dialogs/FaceEditSheet.h"
+
+#include <QHBoxLayout>
+#include <QVBoxLayout>
+#include <QGridLayout>
+#include <QGroupBox>
+#include <QPushButton>
+#include <QDoubleValidator>
+#include <QFormLayout>
+
+using namespace ui;
+
+CFaceEditSheet::CFaceEditSheet( QWidget *pParent ) :
+	QDialog( pParent )
+{
+	this->setWindowTitle( tr( "Face Edit Sheet" ) );
+
+	auto pDialogLayout = new QVBoxLayout( this );
+
+	auto pFaceEditTabWidget = new QTabWidget( this );
+	pFaceEditTabWidget->addTab( new CFaceEditMaterialTab( this ), tr( "Material" ) );
+	pFaceEditTabWidget->addTab( new CFaceEditDisplacementTab( this ), tr( "Displacement" ) );
+
+	pDialogLayout->addWidget( pFaceEditTabWidget );
+
+	this->setLayout( pDialogLayout );
+
+	this->setFixedSize( this->sizeHint() );
+}
+
+//-----------------------------------------------------------------------------------------//
+
+CFaceEditMaterialTab::CFaceEditMaterialTab( QWidget *pParent ) :
+	QWidget( pParent )
+{
+	auto pWidgetLayout = new QGridLayout( this );
+	
+	int row = 0;
+
+#pragma region Texture Scale and Shift
```

Comment:
Not a fan of the pragmas here, would prefer if they were removed.

I think you're better off creating methods for each of these regions to keep things more organized. Large constructors can be kinda crappy